### DISCUSSION TITLE
Control horizon option.

### DISCUSSION THREAD

**Discussion by davidzanger** [do-mpc/do-mpc#324] [Q&A] [answered]
It would be really nice to have a control horizon option for the MPC in order to reduce the number of optimization variables while keeping the same prediction horizon.

**Comment by 4flixt** [comment, marked as answer]
Dear @davidzanger,

thank you for the question/suggestion. Since do-mpc is based on orthogonal collocation to discretize the continuous-time ODE we anyways introduce all states of the horizon (and for all collocation points) as optimization variables. The complexity of the problem would therefore only reduce very mildly by having a different control and prediction horizon as we would only safe on a couple of control inputs as optimization variables. We gain much more from having these additional control inputs to obtain a better control performance.
We also don't see the control vs. prediction horizon in practice very often anymore and consider it (mostly) an approach of the past that will (most likely) not be implemented in do-mpc.

Best,
4flixt
